src/main/kotlin/commands/Nick.kt

```diff
@@ -57,16 +62,70 @@ class Nick(private val chattORE: ChattORE) : BaseCommand() {
         // Note: worth having a timeout to prevent people from changing too frequently.
         if (colors.isEmpty()) throw ChattoreException("No colors provided! Please provide 1 to 3 colors!")
         val rendered = if (colors.size == 1) {
-            val code = "<${colors.first().validateColor()}>"
-            val nickname = "${code}${player.username}<reset>"
+            val color = colors.first().validateColor();
+            val nickname = "<color:$color><username></color:$color>"
             chattORE.database.setNickname(player.uniqueId, nickname)
             nickname
         } else {
             if (colors.size > 3) throw ChattoreException("Too many colors!")
-            setNicknameGradient(player.uniqueId, player.username, *colors)
+            setNicknameGradient(player.uniqueId, *colors)
         }
         val response = chattORE.config[ChattORESpec.format.chattore].render(
-            "Your nickname has been set to $rendered".miniMessageDeserialize()
+            "Your nickname has been set to $rendered".render(mapOf(
+                "username" to Component.text(player.username)
+            ))
+        )
+        player.sendMessage(response)
+    }
+
+    @Subcommand("preset")
+    @CommandCompletion("@nickPresets")
+    fun pride(player: Player, preset: String) {
+        val format = chattORE.config[ChattORESpec.nicknamePresets][preset]
+            ?: throw ChattoreException("Unknown preset! Use /nick presets to see available presets.")
+        val rendered = format.render(mapOf("username" to Component.text(player.username)));
+        chattORE.database.setNickname(player.uniqueId, format);
+        val response = chattORE.config[ChattORESpec.format.chattore].render(
+            "Your nickname has been set to <message>".render(rendered)
+        )
+        player.sendMessage(response)
+    }
+
+    @Subcommand("presets")
+    @CommandCompletion("@username")
+    fun presets(player: Player, @Optional shownText: String?) {
+        var renderedPresets = ArrayList<Component>()
+        for ((presetName, preset) in chattORE.config[ChattORESpec.nicknamePresets]) {
+            val applyPreset: (String) -> Component = {
+                preset.render(mapOf(
+                    "username" to Component.text(it)
+                ))
+            }
+            val rendered = if (shownText == null) {
+                // Primarily show the preset name, else a preview of the nickname.
+                applyPreset(presetName).hoverEvent(
```

Comment:
It may be cleaner to have the [hover event](https://docs.advntr.dev/minimessage/format.html#hover) and [click event](https://docs.advntr.dev/minimessage/format.html#click) defined in MiniMessage.

Reply:
I'm not sure i agree it's "cleaner" but i don't have a particularly strong opinion on either way. I didn't realize this was possible through MiniMessage; i simply followed the [same approach as you use in the sender field of messages](https://github.com/OpenRedstoneEngineers/ChattORE/blob/26fb2d11cab81ef6ee7a3088fb52fde5fee0d8c1/src/main/kotlin/ChattORE.kt#L306-L310)

Reply:
To be honest I forget why I didn't use MiniMessage there.

Reply:
I could rewrite both of these to use MiniMessage format, then.

Reply:
bonus feature unlocked: hovertext of usernames is now configurable easily.

i don't feel like the click events benefit much from being defined in MiniMessage, but yeah the hover events look nicer now